formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -196,4 +195,19 @@ internal class FormulaManagerImpl<Input, State : Any, Output>(
             key = formula.key(input)
         )
     }
+
+    private fun getOrInitChildTransitionListener(): TransitionListener {
+        return childTransitionListener ?: run {
```

Comment:
You can return the `TransitionListener` object directly, probably more Kotlin idiomatic:
```
        return childTransitionListener ?: run {
            TransitionListener { transition, isValid ->
                val frame = this.frame
                if (!isValid) {
                    frame?.childInvalidated()
                }
                val isValid = frame != null && frame.isValid()
                transitionListener.onTransition(transition, isValid)
            }.apply {
                childTransitionListener = this
            }
        }
```

Reply:
I was debating if I should use this syntax.